zulipterminal/ui_tools/views.py

```diff
@@ -1052,16 +1053,19 @@ def make_table_with_categories(
     def keypress(self, size: urwid_Size, key: str) -> str:
         if is_command_key("GO_BACK", key) or is_command_key(self.command, key):
             self.controller.exit_popup()
-        elif is_command_key("GO_UP", key):
-            key = "up"
-        elif is_command_key("GO_DOWN", key):
-            key = "down"
-        elif is_command_key("SCROLL_UP", key):
-            key = "page up"
-        elif is_command_key("SCROLL_DOWN", key):
-            key = "page down"
-        elif is_command_key("GO_TO_BOTTOM", key):
-            key = "end"
+        # Disable navigational commands when inside search-box to
+        # allow typing secondary letter-keys.
+        if not self.is_popup_search_open:
```

Comment:
A better way to check this is using `self.controller.is_in_editor_mode`, but the issue is using this would Fail a lot of other tests that would by-default assume `.is_in_editor_mode` to be `Mock` object and the above condition would always be `False`, preventing a lot of navigation keypress tests from passing.

I am having difficulty figuring out if I should go ahead with the current approach, or use the above one. The current approach as @Rohitth007 notes [here](https://github.com/zulip/zulip-terminal/pull/913/commits/f05c89246cbd529c1a8d12a8a0e5b645f6335866#r650673634), kills the abstraction of class. But, the above approach would require modifying un-related tests.

Reply:
Would calling `super().keypress` in EmojiPickerView only when not in search mode be a simpler solution?

Reply:
If we dont call `super.keypress()` when in search mode and just return, then none of the keys would get typed in the search-box, right?
We need to handle this in `super` only, as far as I can think.

Reply:
I am going forward with the current approach since that doesn't involve unnecessary test changes, leading to an even larger PR. But, I am keeping this conversation open for further discussion. :)